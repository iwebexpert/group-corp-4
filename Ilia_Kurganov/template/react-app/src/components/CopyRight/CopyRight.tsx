import React from 'react'
import { Link, Typography, TypographyProps } from '@mui/material'

export const CopyRight = ( props: TypographyProps ) => {
  return (
    <Typography variant="body2" color="text.secondary" align="center" {...props}>
      {'Copyright Â© '}
      <Link color="inherit" href="https://mui.com/">
        Test Project
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  )
}
