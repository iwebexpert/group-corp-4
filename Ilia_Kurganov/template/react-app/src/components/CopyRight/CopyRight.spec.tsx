import React from 'react'

import { CopyRight } from './CopyRight'

import * as renderer from 'react-test-renderer'
import { test, expect } from '@jest/globals'
import  Link  from '@mui/material/Link'
import { shallow, mount } from 'enzyme'


test('Copyright should be rendere', () => {
  // const a = 3
  // expect(a).toBe(3)

  const testRenderer = renderer.create(<CopyRight />)
  const treeJson = testRenderer.toJSON()
  expect(treeJson).toMatchSnapshot()

  const testInstance = testRenderer.root
  expect(testInstance.findByType(Link).props.href).toBe('https://mui.com/')
})

test('Copyright should be rendere with enzyme', () => {

  // const tree = shallow(<CopyRight />)
  // const a = tree.find('ForwardRef(Typography)')

  const tree = mount(<CopyRight />)
  const a = tree.find('a')

  // console.log(tree.debug())

  expect(a.length).toBe(1)
})
