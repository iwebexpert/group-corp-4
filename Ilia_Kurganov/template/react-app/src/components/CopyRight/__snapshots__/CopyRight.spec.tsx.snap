// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright should be rendere 1`] = `
<p
  className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
>
  Copyright Â© 
  <a
    className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1kgfegd-MuiTypography-root-MuiLink-root"
    href="https://mui.com/"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    Test Project
  </a>
   
  2022
  .
</p>
`;
